### PR TITLE
[Routing] Add backslashes in front of global functions in the generated matcher

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Just wondering @frankdejonge: can you spot any performance benefits doing this on PHP7 with your benchs (since you did some recently?)

### CODE REVIEW

File: src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ private function compileRoutes(RouteCollection $routes, $supportsRedirections)
                     $fetchedHost = true;
                 }
 
-                $code .= sprintf("        if (preg_match(%s, \$host, \$hostMatches)) {\n", var_export($regex, true));
+                $code .= sprintf("        if (\\preg_match(%s, \$host, \$hostMatches)) {\n", var_export($regex, true));
```
Comment (curry684) [inline src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php]: Any specific reason for skipping `var_export` consistently?

**Comment by GromNaN** [comment]
In the standard edition, the class is generated in the global namespace.

**Comment by nicolas-grekas** [comment]
> In the standard edition, the class is generated in the global namespace.

Indeed, that's the killer arg to close this PR, thanks :)

**Comment by frankdejonge** [comment]
@nicolas-grekas for future reference, I used this: https://gist.github.com/frankdejonge/0ab1b9d386f7393e151538f0486ddd18
But profiling requests in blackfire provides a far nicer insight into where the speed is improved. If you need help with that for routing stuff, lemme know and I'll see how/where/when I can help out.

**Pull request closed without merging**